src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -248,49 +237,59 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
         /// Workaround for interactive window that does not currently support
         /// 'carriage return' i.e. writing into the same line
         /// </summary>
-        private volatile string _lastMessage;
-        private int _lastPosition = -1;
+        struct MessagePos {
+            public string Message;
+            public int Position;
+        }
 
+        private readonly Stack<MessagePos> _messageStack = new Stack<MessagePos>();
```

Comment:
So in case we do get multiple changes they will be  applied in reverse order so positions won't change for change 1 after change 2 is applied.